rclpy/test/test_qos_event.py

```diff
@@ -0,0 +1,238 @@
+        callbacks = PublisherEventCallbacks()
+        liveliness_callback = Mock()
+        deadline_callback = Mock()
+
+        # No arg
+        publisher = self.node.create_publisher(EmptyMsg, 'test_topic')
+        self.assertEqual(len(publisher.event_handlers), 0)
+        self.node.destroy_publisher(publisher)
+
+        # Arg with no callbacks
+        publisher = self.node.create_publisher(EmptyMsg, 'test_topic', event_callbacks=callbacks)
+        self.assertEqual(len(publisher.event_handlers), 0)
+        self.node.destroy_publisher(publisher)
+
+        # Arg with one of the callbacks
+        callbacks.deadline = deadline_callback
+        publisher = self.node.create_publisher(EmptyMsg, 'test_topic', event_callbacks=callbacks)
+        self.assertEqual(len(publisher.event_handlers), 1)
+        self.node.destroy_publisher(publisher)
+
+        # Arg with both callbacks
+        callbacks.liveliness = liveliness_callback
+        publisher = self.node.create_publisher(EmptyMsg, 'test_topic', event_callbacks=callbacks)
+        self.assertEqual(len(publisher.event_handlers), 2)
+        self.node.destroy_publisher(publisher)
+
+    def test_subscription_constructor(self):
+        callbacks = SubscriptionEventCallbacks()
+        liveliness_callback = Mock()
+        deadline_callback = Mock()
+        message_callback = Mock()
+
+        # No arg
+        subscription = self.node.create_subscription(EmptyMsg, 'test_topic', message_callback)
+        self.assertEqual(len(subscription.event_handlers), 0)
+        self.node.destroy_subscription(subscription)
+
+        # Arg with no callbacks
+        subscription = self.node.create_subscription(
+            EmptyMsg, 'test_topic', message_callback, event_callbacks=callbacks)
+        self.assertEqual(len(subscription.event_handlers), 0)
+        self.node.destroy_subscription(subscription)
+
+        # Arg with one of the callbacks
+        callbacks.deadline = deadline_callback
+        subscription = self.node.create_subscription(
+            EmptyMsg, 'test_topic', message_callback, event_callbacks=callbacks)
+        self.assertEqual(len(subscription.event_handlers), 1)
+        self.node.destroy_subscription(subscription)
+
+        # Arg with both callbacks
+        callbacks.liveliness = liveliness_callback
+        subscription = self.node.create_subscription(
+            EmptyMsg, 'test_topic', message_callback, event_callbacks=callbacks)
+        self.assertEqual(len(subscription.event_handlers), 2)
+        self.node.destroy_subscription(subscription)
+
+    def test_publisher_event_create_destroy(self):
+        # Publisher event types
+        publisher = self.node.create_publisher(EmptyMsg, 'test_topic')
+
+        event_handle = _rclpy.rclpy_create_event(
+            QoSPublisherEventType.RCL_PUBLISHER_OFFERED_DEADLINE_MISSED,
+            publisher.publisher_handle)
+        self.assertIsNotNone(event_handle)
+        _rclpy.rclpy_destroy_entity(event_handle)
```

Comment:
Please wrap in `rclpy.handle.Handle` and call `Handle.destroy()`. This function will be removed shortly

Reply:
ok